tortoisebeacon/weakcoin/handler.go

```diff
@@ -0,0 +1,67 @@
+package weakcoin
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/p2p/service"
+)
+
+// HandleSerializedMessage defines method to handle Tortoise Beacon Weak Coin Messages from gossip.
+func (wc *WeakCoin) HandleSerializedMessage(ctx context.Context, data service.GossipMessage, sync service.Fetcher) {
+	wc.logger.With().Debug("received weak coin message",
+		log.String("from", data.Sender().String()),
+		log.Binary("data", data.Bytes()),
+	)
+
+	var message Message
+	if err := types.BytesToInterface(data.Bytes(), &message); err != nil {
+		wc.logger.With().Debug("received invalid weak coin message",
+			log.Binary("message", data.Bytes()),
+			log.Err(err))
+		return
+	}
+
+	if err := wc.receiveMessage(message); err != nil {
+		wc.logger.With().Debug("received invalid proposal",
+			log.Err(err),
+			log.Uint32("epoch_id", uint32(message.Epoch)),
+			log.Uint64("round_id", uint64(message.Round)))
+		return
+	}
+	data.ReportValidation(ctx, GossipProtocol)
+}
+
+func (wc *WeakCoin) receiveMessage(message Message) error {
+	if wc.exceedsThreshold(message.Signature) {
+		return fmt.Errorf("proposal %x exceeds threshold", message.Signature)
```

Comment:
As you're touching this, could you please negate the logic of this function and rename it to `passesThreshold`? There was the same change in the `tortoisebeacon` package: https://github.com/dshulyak/go-spacemesh/blob/finalize-weakcoin/tortoisebeacon/tortoise_beacon.go#L942

```suggestion
	if !wc.passesThreshold(message.Signature) {
		return fmt.Errorf("proposal %x doesn't pass threshold", message.Signature)
```

Reply:
i will rename but i don't think that it matters how simple helper is called in two different places

Reply:
@nkryuchkov or do you mean that the signature must be above threshold now? should be strictly above or above or equal?

Reply:
@dshulyak If the signature is strictly below the threshold, it passes the threshold and needs to be handled. If it's equal or above the threshold, it doesn't pass the threshold and should be rejected

Reply:
i think it is very confusing to call it passes, when it actually means that it verifies that it actually needs to be below the threshold. exceeds is already better, but it can be named even if `aboveThreshold() { return err }`

Reply:
It was a request from the research team to call it "passes". But we can leave it as "exceeds"